### PR TITLE
Fix macOS BigSur configure problem finding pthreads

### PR DESCRIPTION
**Pull request by pathron**
This aims to fix the problem here https://github.com/FlexibleSUSY/FlexibleSUSY/issues/292 .

@azedarach helped find the right use of sed etc on the macOS, but I will need to get a BigSur user to test it before we can merge it in (i will see if my GAMBIT collaborator Jonathan Cornell can after he has tested a similar fix in GAMBIT).  In the meantime Alex will probably want to look at the very ugly way I do this, though and may have comments.

Finally it only fixes the issue of not finding the library.  For the FS - 2.0.1 in gambit I have also made a change so that if the pthread libs are not found it simply disables multi-threading (as long as it was not explicitly enabled) instead of always giving a configure error.  I think we should do something similar in FS but the configure script has evolved so much its quite a different change that is is needed to do this.  For 2.0.1  the changes were like in this diff code:

```
diff --git a/contrib/MassSpectra/flexiblesusy/configure b/contrib/MassSpectra/flexiblesusy/configure
index b9cbae8..97c1f8a 100755
--- a/contrib/MassSpectra/flexiblesusy/configure
+++ b/contrib/MassSpectra/flexiblesusy/configure
@@ -1276,7 +1276,6 @@ check_std_threads() {
         if try_compile_cpp_program "#include <thread>
 #include <mutex>" "std::mutex mtx; std::thread thr;";
         then
-            enable_threads="yes"
             result "ok"
         else
             result "not available"
@@ -1297,11 +1296,23 @@ check_std_threads() {
 check_thread_libs() {
     # link libpthread only if $enable_threads = yes
     # I'm assuming here, that no other library we link needs libpthread.
-    if [ "x$enable_threads" = "xyes" -o -n "$pthread_lib_dir" ] ; then
+    if [ "x$enable_threads" = "xno" ] ; then
+	return 0
+    fi
+    if [ -n $pthread_lib_dir ] ; then
+	 message "Checking for libpthread..."
         check_library "libpthread" "$pthread_lib_dir" "$default_lib_paths"
-        if [ -z "$found_lib" ]; then
-            message "Error: libpthread must be installed to enable multithreading!"
-            exit 1
+        if [ -z "$found_lib" ] ; then
+            message "cannot find libpthread, which needs to be installed to enable multithreading."
+            if test "x${enable_threads}" = "xyes" ; then
+		message "Either reconfigure with \`--disable-threads' or"
+		message "pass the library location with \`--with-pthread-libdir='"
+		exit 1
+	    fi
+	    message "will disable multi-threading"
+	    enable_threads="no"
+	    return 0
         fi
+	enable_threads="yes"
         THREADLIBS=-l$(echo "$found_lib" | sed 's/^lib\(.*\)\..*$/\1/')
         # need to link whole libpthread.a for static linking
         if [ "x$enable_static" = xyes ] ; then
```

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -1518,10 +1518,10 @@ check_thread_libs() {
 
     # additional search paths by the compiler
     local cc_lib_search_paths=""
-
     case "$cxx_compiler_type" in
 	gnu|clang)
 	    cc_lib_search_paths=$(${CXX} -print-search-dirs | sed -n -e '/libraries:/s/libraries: *=//p')
+	    cc_lib_search_paths=$cc_lib_search_paths$(${CXX} -Xlinker -v 2>&1 | sed -n '/Library search paths:/,/Framework search paths:/p' | sed '/Library search paths:/d; /Framework search paths:/d' | sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/ /g' | tr -s '[:blank:]' ':')
```
Comment (Expander) [inline configure:1524]: I would recommend to replace `2>&1` by `2>/dev/null`, because we don't need to parse the error output.

**Comment by Expander** [comment]
Hi Peter, many thanks for the patch!

I've found another approach at https://stackoverflow.com/questions/9922949/how-to-print-the-ldlinker-search-path :

~~~
g++ -Xlinker --verbose  2>/dev/null | grep SEARCH | sed 's/SEARCH_DIR("=\?\([^"]\+\)"); */\1\n/g'  | grep -vE '^$'
~~~

It uses `g++ -Xlinker --verbose  2>/dev/null | grep SEARCH` and extracts all paths from the `SEARCH_DIR` expression. I'm wondering if this is an alternative? On my Debian/Linux machine it prints:
~~~
$ g++ -Xlinker --verbose  2>/dev/null | grep SEARCH | sed 's/SEARCH_DIR("=\?\([^"]\+\)"); */\1\n/g'  | grep -vE '^$'
/usr/local/lib/x86_64-linux-gnu
/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu64
/usr/local/lib64
/lib64
/usr/lib64
/usr/local/lib
/lib
/usr/lib
/usr/x86_64-linux-gnu/lib64
/usr/x86_64-linux-gnu/lib
~~~

File: configure (hunk 1 above)
Comment (pathron) [reply configure:1524]: I don't think that works.  Commands with that were failing and Dylan told me that the entire output is error so do need tp parse it.

**Comment by pathron** [comment]
Sorry, but I already looked at that page and tried commands like the one you propose. It works fine on ubuntu, but the mac output from `g++ -Xlinker --verbose` does not give the output path you need. In fact I think it does not give the output you are trying to parse there at all.  That is why I have spent the last month or so going back and forth with Jonathan...

File: configure (hunk 1 above)
Comment (Expander) [reply configure:1524]: Ok, then we must leave it.

**Comment by Expander** [comment]
Ok, thanks for checking! So, your patch is basically one that is well tested on MacOS. I think then that is fine for me.

**Comment by pathron** [comment]
So Jonathan just confirmed this worked for him, so i think it should be good.

However its showing some failing tests conan and test/test_flexiblesusy-config.sh .

**Comment by Expander** [comment]
> However its showing some failing tests conan and test/test_flexiblesusy-config.sh .

That's strange, because the `development` branch passes. Is this branch maybe outdated?

**Comment by pathron** [comment]
Well it was up to date when I made the pull request, but I can merge in the latest and see if that helps.

**Comment by pathron** [comment]
I forget to check this after merging and pushing, but all tests are passing now so I think I will go ahead and merge.

**Pull request merged**